### DISCUSSION TITLE
GUI like Tkinter with PyAppp, possible?

### DISCUSSION THREAD

**Discussion by allomorphy** [ofek/pyapp#59] [Q&A]
Thank you @ofek 
We are super keen to try this out.
I am wondering if anyone has successfully PyApp'ed a GUI?

Warm wishes
Eric

**Comment by ofek** [comment]
I'm not familiar with GUIs but I would have assumed it's the same process, no?

**Reply by allomorphy** [reply]
Me neither, but thanks Ofek. When I get there, I will let you know how it goes!

**Comment by trappitsch** [comment]
It actually works pretty great! I put together a very simple PyQt6 app [here](https://github.com/trappitsch/qtcowsay) and then used the following steps with `pyapp`:

- Download the latest version as described in the [how-to](https://ofek.dev/pyapp/latest/how-to/)
- Create a `requirements.txt` pointing at `pip` installation from git, e.g., via
  ```bash
  echo 'git+https://github.com/trappitsch/qtcowsay' > requirements.txt
  ```
- Set the following environment variables
  | Option                        | Value              |
  |-------------------------------|--------------------|
  | PYAPP_PROJECT_NAME            | qtcowsay           |
  | PYAPP_PROJECT_VERSION         | 0.0.2              |
  | PYAPP_PROJECT_DEPENDENCY_FILE | ./requirements.txt |
  | PYAPP_EXEC_SPEC               | qtcowsay:run       |

Then simply `cargo build --release` and the GUI should start.

Just my first test, still experimenting with local setup, isolation, and some nicer entry into the module... (the `__init__.py` is currently kind of ugly and just runs the app upon import...).

Hope this helps!

**Updated**: see comment below

**Comment by trappitsch** [comment]
I played a bit more and got the local installation to work and updated the [repo mentioned above](https://github.com/trappitsch/qtcowsay). There's more info in the `README` on the repo site.

I also updated the project a bit, sorry. Will update above instructions in a second to represent the changes.
